zpm/publisher_abs.go

```diff
@@ -397,20 +435,42 @@ func (a *ABSPublisher) publish(osarch *zps.OsArch, pkgFiles []string, zpkgs []*z
 
 	locker := NewLocker(a.lockUri)
 
-	err = locker.Lock()
+	eTag, err := locker.LockWithEtag()
 	if err != nil {
 		return fmt.Errorf("repository: %s is locked by another process, error: %s", a.name, err.Error())
 	}
 
-	defer locker.Unlock()
+	defer locker.UnlockWithEtag(newEtag)
```

Comment:
This needs to be `defer locker.Unlock()`, and `defer.UnlockWithEtag(newTag)` should be called with the happy path at the bottom, once the new etag has been created.

This applies to all code using defer for unlocking.

Reply:
see comment above